egs/wsj/s5/steps/nnet3/train_rnn.py

```diff
@@ -0,0 +1,1134 @@
+                        be run locally or in the cluster.""", default = "cluster",
+                        choices = ["cluster", "local"])
+    parser.add_argument("--use-gpu", type=str,
+                        help="Use GPU for training", default="true",
+                        choices = ["true", "false"])
+    parser.add_argument("--cleanup", type=str,
+                        help="Clean up models after training", default="true",
+                        choices = ["true", "false"])
+    parser.add_argument("--max-lda-jobs", type=int,
+                        help="Number of jobs for LDA accumulation.", default=10)
+
+    parser.add_argument("data", type=str, 
+                        help="Directory with features used for training the neural network.")
+    parser.add_argument("lang", type=str,
+                        help="Languade directory")
+    parser.add_argument("alidir", type=str,
+                        help="Directory with alignments used for training the neural network.")
+    parser.add_argument("dir", type=str,
+                        help="Directory to store the models and all other files.")
+
+    print(' '.join(sys.argv))
+
+    args = parser.parse_args()
+
+
+    # process the options
+    if args.use_gpu == 'true':
+        args.use_gpu = True
+    else:
+        args.use_gpu = False
+
+    if args.cleanup == 'true':
+        args.cleanup = True
+    else:
+        args.cleanup = False
+
+    if args.chunk_width < 1:
+        raise Exception("--chunk-width should have a minimum value of 1")
+
+    if args.chunk_left_context < 0:
+        raise Exception("--chunk-left-context should be positive")
+
+    if args.chunk_right_context < 0:
+        raise Exception("--chunk-right-context should be positive")
+
+    if not os.exists(args.dir):
+        raise Exception("This scripts expects {0} to exist and have a configs
+        directory which is the output of make_configs.py script")
+
+
+    # set the options corresponding to args.use_gpu
+    run_opts = RunOpts()
+    if args.use_gpu:
+        if not CheckIfCudaCompiled():
+            logger.warning("""
+    You are running with one thread but you have not compiled
+    for CUDA.  You may be running a setup optimized for GPUs.  If you have
+    GPUs and have nvcc installed, go to src/ and do ./configure; make""")
+
+        run_opts.train_queue_opt = "--gpu 1"
+        run_opts.parallel_train_opts = ""
+        run_opts.combine_queue_opt = "--gpu 1"
+        run_opts.prior_gpu_opt = "--use-gpu=yes"
+        run_opts.prior_queue_opt = "--gpu 1"
+
+    else:
+        logger.warning("""
+    Without using a GPU this will be very slow.  nnet3 does not yet support multiple threads.""")
+        
+        run_opts.train_queue_opt = ""
+        run_opts.parallel_train_opts = "--use-gpu=no"
+        run_opts.combine_queue_opt = ""
+        run_opts.prior_gpu_opt = "--use-gpu=no"
+        run_opts.prior_queue_opt = ""
+    
+    if args.run_in == "cluster":
```

Comment:
Vijay, this destroys all the power of the $cmd variable, just having two options like that, and it wouldn't be compatible with existing scripts.  It's not a rule that $cmd can only be one of two options-- may people use their own scripts.